crates/extensions/c8y_config_manager/src/config.rs

```diff
@@ -95,6 +95,23 @@ impl ConfigManagerConfig {
             mqtt_port,
             tedge_http_address,
             tedge_http_port,
-        ))
+        );
+        is_config_dir_exist(config.plugin_config_path.as_path())?;
```

Comment:
Rather than creating a "faulty" config object first and then doing an explicit validation with this extra function call, why not make the `ConfigManagerConfig::new()` (`ConfigManagerConfig::try_new()`) return the error?

Reply:
Many `new` functions must be converted to `try_new`, because it's a bit deeper in the code(c8y_config_plugin->plugin-config->RawPluginConfig). So, this is the right way to validate the config path.

Reply:
I agree with @albinsuresh.

> Many new functions must be converted to try_new, 

There are only 2 calls to `ConfigManagerConfig::new()` ;-)